dcpy/test/connectors/edm/test_publishing.py

```diff
@@ -1,10 +1,49 @@
 import pytest
+import pandas as pd
 from dcpy.connectors.edm import publishing
 
-TEST_PRODUCT_NAME = "test_product"
-TEST_VERSION = "version_001"
+TEST_PRODUCT_NAME = "test-product"
+TEST_BUILD = "build-branch"
+TEST_VERSION = "v001"
+TEST_ACL = "bucket-owner-read"
 
+draft_key = publishing.DraftKey(product=TEST_PRODUCT_NAME, build=TEST_BUILD)
+publish_key = publishing.PublishKey(product=TEST_PRODUCT_NAME, version=TEST_VERSION)
 
-@pytest.fixture
-def get_draft_key(scope="function"):
-    return publishing.DraftKey(product=TEST_PRODUCT_NAME, build=TEST_VERSION)
+
+def test_bucket_empty(create_bucket, mock_data_constants):
```

Comment:
It looks like these tests depend on each other? (And specifically, they have to run in specific order?) It's typically an anti-pattern in unit testing, though given our setup I don't think it will cause problems given that we run tests serially. But you typically want to ensure that each `test_` method is actually a *unit* without implicit dependencies on other units. Put another way, you'd want to write all unit tests so that they can be run in parallel.

Reply:
@alexrichey, good point! I didn't notice the dependency among the tests. I would like to address it and here is how I'm thinking of going about it: 

- revise fixtures' scope to `function`
- for `test_publish()`, call `publishing.upload` fn in the beginning (`publishing.upload` fn is tested in `test_upload`).

What do you think?

Reply:
@alexrichey, yes, that's what I was thinking of...